sdk/core/azure-core/inc/azure/core/azure_assert.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Provide assert macros to use as pre-conditions.
+ *
+ * @remark Asserts are turned ON even if `NDEBUG` is defined.
+ *
+ */
+
+#pragma once
+
+#if defined(NDEBUG)
+#define AZURE_SDK_INTERNAL_IMPLEMENTATION_OVERRIDE_NDEBUG
+#undef NDEBUG
+#endif
```

Comment:
This approach seems very brittle to me. It's easily broken by normal user code (https://godbolt.org/z/nMK4nxc9n). (Indicates lack of test coverage.)